email/smtpDriver.go

```diff
@@ -0,0 +1,128 @@
+package email
+
+import (
+    "net"
+	"net/smtp"
+    "crypto/tls"
+    "strconv"
+    "github.com/grubastik/flat-search/config"
+)
+
+const configName = "email"
+
+type EmailConnection struct {
+    connection *smtp.Client
+    server string
+    port int
+    tlsEnabled bool
+    username string
+    password string
+}
+
+var emailConf *config.Email
+
+var Conn *EmailConnection
+
+func NewEmailConnection(config *config.Config) *EmailConnection {
+    var ec *EmailConnection = new(EmailConnection);
+    emailConf = config.GetEmail();
+
+    if len(emailConf.Server) > 0 {
+        ec.server = emailConf.Server;
+    }
+    if emailConf.TlsPort > 0 {
+        ec.port = emailConf.TlsPort;
+    }
+    if emailConf.Tls {
+        ec.tlsEnabled = emailConf.Tls;
+    }
+    if len(emailConf.Username) > 0 {
+        ec.username = emailConf.Username;
+    }
+    if len(emailConf.Password) > 0 {
+        ec.password = emailConf.Password;
+    }
+    return ec;
+}
+
+func (ec *EmailConnection) Send(ed *EmailDefinition) error {
+    // Setup message
+    message := ed.getMessage();
+
+    ec.makeSmtpClient()
```

Comment:
It's kind of overhead create client on each `Send` action. Init your client/open connection and keep that client in memory. What if you need to send 100000 emails? With current logic, you need to connect to smtp server upon sending each email.